services/connectivity/messaging/src/main/java/org/eclipse/ditto/services/connectivity/messaging/amqp/AmqpPublisherActor.java

```diff
@@ -108,8 +107,21 @@ public AmqpPublisherActor create() {
 
     @Override
     protected void preEnhancement(final ReceiveBuilder receiveBuilder) {
-        // noop
+        receiveBuilder.match(StatusReport.class, this::handleConnectionStatusReport);
     }
+    
+    private void handleConnectionStatusReport(final StatusReport report) {
+      report.getClosedProducer().ifPresent(producer -> {  
+              log.debug("Try to remove JMS producer '{}' from cache.", producer);
+              Optional<Map.Entry<Destination, MessageProducer>> remove = producerMap.entrySet().stream().
```

Comment:
I know this is ugly btw. However, when closed MessageProducer.getDestination is rejecting access as well. So I had to do this brute-force.